### ISSUE TITLE
Opening a project scene AN crhases - Problem with "inactive" collections detected

### ISSUE THREAD

**Issue by juangea** [JacquesLucke/animation_nodes#1110]
So after building it, it seemed to work, but something is happening, if I open a new scene I can use AN, but when I open an old scene (with old I mean a day or two) that is more complex I get this crash and it does not work:

![image](https://user-images.githubusercontent.com/20524718/58388015-fafd8580-8018-11e9-9c74-b5972c2458a8.png)
```
Traceback (most recent call last):
  File "/home/jgea/.config/blender/2.80/scripts/addons/animation_nodes/utils/handlers.py", line 44, in always
    handler()
  File "/home/jgea/.config/blender/2.80/scripts/addons/animation_nodes/utils/selection.py", line 20, in updateSelectionSorting
    selectedNames = getSelectedObjectNames()
  File "/home/jgea/.config/blender/2.80/scripts/addons/animation_nodes/utils/selection.py", line 39, in getSelectedObjectNames
    return [obj.name for obj in bpy.context.scene.objects if obj.select_get()]
  File "/home/jgea/.config/blender/2.80/scripts/addons/animation_nodes/utils/selection.py", line 39, in <listcomp>
    return [obj.name for obj in bpy.context.scene.objects if obj.select_get()]
RuntimeError: Error: Object 'suelo_cesped' not in View Layer 'View Layer'!

```

Some ideas?

**Comment by juangea** [comment]
I get that error everytime I open that scene, and I don't know why, I tried everything but nothing seems to achieve Animation Nodes not crashing :P

**Comment by juangea** [comment]
no way to solve it, I tried to open the scene with AN disabled, as soon as I enable it it gives me this error:
```
Registered Animation Nodes
Traceback (most recent call last):
  File "/home/jgea/.config/blender/2.80/scripts/addons/animation_nodes/utils/handlers.py", line 44, in always
    handler()
  File "/home/jgea/.config/blender/2.80/scripts/addons/animation_nodes/utils/selection.py", line 20, in updateSelectionSorting
    selectedNames = getSelectedObjectNames()
  File "/home/jgea/.config/blender/2.80/scripts/addons/animation_nodes/utils/selection.py", line 39, in getSelectedObjectNames
    return [obj.name for obj in bpy.context.scene.objects if obj.select_get()]
  File "/home/jgea/.config/blender/2.80/scripts/addons/animation_nodes/utils/selection.py", line 39, in <listcomp>
    return [obj.name for obj in bpy.context.scene.objects if obj.select_get()]
RuntimeError: Error: Object 'planta_urba' not in View Layer 'View Layer'!

```

**Comment by juangea** [comment]
Ok, I found the culptrip!!!!

If I have any collection disabled (actual really disabled) it gives the error, with disabled I mean this checkbox:

![image](https://user-images.githubusercontent.com/20524718/58388485-f7b9c800-801f-11e9-88c2-4036bdad11a4.png)

**Comment by OmarSquircleArt** [comment]
Thanks for the report! Will look into it

**Comment by juangea** [comment]
I think it's a matter of the recetn API changes, let me know when you manage to fix it so I can re-build AN because that's an important feature right now, helps a lot to keep scenes clean :)

**Comment by Clockmender** [comment]
I don't have this latest version of Blender with the "Disable Collections" checkbox, but the offending line:

`return [obj.name for obj in bpy.context.scene.objects if obj.select_get()]`

Could be replaced with:

`return [obj for obj in bpy.context.view_layer.objects if obj.visible_get() and obj.select_get()]`

As I have it in my own version of this, which works in my Blender version dated 2019-04-05. This only adds _displayed_ objects, not all scene objects. I am presuming disabled collections are not visible.

Just a thought...

**Comment by OmarSquircleArt** [comment]
Thanks @Clockmender and @juangea ! Fixed in 1f9db1c.
